scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,316 @@
+        for proc_type in PROCESS_TYPES:
+            usage_str[proc_type] = get_usage_str(fault, proc_type)
+            ncl = mgmtdb.num_task_complete(
+                (const.ProcessType[proc_type].value, fault + "_REL%"), like=True
+            )
+            total_complete_rels[proc_type] = total_complete_rels.get(proc_type, 0) + ncl
+            num_complete_rels[proc_type] = f"{ncl}/{num_all_rels}"
+            is_complete[proc_type] = 1 if ncl == num_all_rels else 0
+
+        output_list = np.array(
+            list(
+                zip(
+                    usage_str.values(), num_complete_rels.values(), is_complete.values()
+                )
+            )
+        ).flatten()
+        f.write(
+            _fault_template_fmt.format(
+                fault, *output_list, get_usage_str(fault, "total")
+            )
+        )
+        f.write("\n")
+
+        for proc_type in PROCESS_TYPES:
+            if is_complete[proc_type]:
+                est_completed_all[fault][proc_type] = progress_df.loc[
+                    fault, (proc_type, EST_CORE_HOURS_COL)
+                ]
+
+    est_completed_df = pd.DataFrame.from_dict(
+        est_completed_all, orient="index", columns=PROCESS_TYPES
+    )
+
+    overall_df = progress_df.sum()
+
+    total_r_count = progress_df["total", R_COUNT_COL].sum()
+
+    f.write("\nOverall progress\n")
+
+    for proc_type in PROCESS_TYPES:
+        f.write(
+            _summary_template_fmt.format(
+                proc_type,
+                total_complete_rels[proc_type],
+                total_r_count,
+                total_complete_rels[proc_type] / total_r_count * 100,
+                overall_df[proc_type, ACT_CORE_HOURS_COL],
+                overall_df[proc_type, EST_CORE_HOURS_COL],
+                overall_df[proc_type, ACT_CORE_HOURS_COL]
+                / overall_df[proc_type, EST_CORE_HOURS_COL]
+                * 100,
+                overall_df[proc_type, EST_CORE_HOURS_COL]
+                - est_completed_df[proc_type].sum(),
+            )
+        )
+    f.close()
+
+
+def send2slack(msg_file, users, url):
+    with open(msg_file, "r") as f:
+        lines = f.readlines()
+
+    summary = (
+        " ".join(users) + " " + "\n".join(lines[-4:])
+    )  # .encode('utf-8') #last 4 lines
+    print(summary)
+    data = json.dumps({"text": summary})
+    urlopen(url, data=data.encode("utf-8"))
+
+
+def main(root_dir: Path, cybershake_list, outfile, slack_config=None):
+
+    runs_dir = Path(sim_struct.get_runs_dir(root_dir))
+    fault_names, r_counts = get_faults_and_r_count(cybershake_list)
+
+    sort_ind = np.argsort(fault_names)
+    fault_names, r_counts = fault_names[sort_ind], r_counts[sort_ind]
+
+    faults = np.asarray(
+        [sim_struct.get_fault_dir(root_dir, fault_name) for fault_name in fault_names]
+    )
+
+    if not root_dir.is_dir() or not runs_dir.is_dir():
+        print("Not a valid cybershake root directory. Quitting!")
+        return
```

Comment:
This check should happen earlier, after the two variables are known to exist, and should raise an exception instead of exiting gracefully. Something's wrong and the user should know about it